crates/bevy_debug_overlays/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+                                color: Color::WHITE,
+                            },
+                        ),
+                        TextSection::new(
+                            " (ms)\n",
+                            TextStyle {
+                                font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                                font_size: 10.0,
+                                color: Color::WHITE,
+                            },
+                        ),
+                    ]));
+                    parent.spawn(NodeBundle::default()).with_children(|parent| {
+                        let style = TextStyle {
+                            font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                            font_size: 12.0,
+                            color: Color::WHITE,
+                        };
+                        parent.spawn((
+                            GpuTimerLabelMarker,
+                            TextBundle::from_section("", style.clone()),
+                        ));
+                        parent.spawn((GpuTimerDurationMarker, TextBundle::from_section("", style)));
+                    });
+                });
+        });
+}
+
+#[derive(Resource, Default)]
+struct AggregatedGpuTimers(HashMap<String, f64>);
+
+// TODO: Handle nesting
+fn aggregate_gpu_timers(
+    gpu_timers: Res<GpuTimerScopes>,
+    mut aggregated_gpu_timers: ResMut<AggregatedGpuTimers>,
+) {
+    let mut stack = gpu_timers.take();
+    while let Some(gpu_timer) = stack.pop() {
+        let average = aggregated_gpu_timers.0.entry(gpu_timer.label).or_default();
+        let duration = gpu_timer.time.end - gpu_timer.time.start;
+        *average = (*average * 0.1) + (duration * 0.9);
+
+        for gpu_timer in gpu_timer.nested_scopes {
+            stack.push(gpu_timer);
+        }
+    }
+}
+
+fn draw_node_gpu_time_overlay(
+    aggregated_gpu_timers: Res<AggregatedGpuTimers>,
+    mut labels: Query<&mut Text, (With<GpuTimerLabelMarker>, Without<GpuTimerDurationMarker>)>,
+    mut durations: Query<&mut Text, (With<GpuTimerDurationMarker>, Without<GpuTimerLabelMarker>)>,
+) {
+    let mut gpu_timers = aggregated_gpu_timers.0.iter().collect::<Vec<_>>();
+    gpu_timers.sort_unstable_by(|(_, d1), (_, d2)| d1.partial_cmp(d2).unwrap().reverse());
+
+    let mut labels = labels.single_mut();
+    let mut durations = durations.single_mut();
+    let labels = &mut labels.sections[0].value;
+    let durations = &mut durations.sections[0].value;
+    labels.clear();
+    durations.clear();
+
+    for (label, duration) in gpu_timers {
+        let label = match label.rsplit_once("::") {
+            Some((_, label)) => label,
+            None => label,
+        };
+        let label = label.strip_suffix("Node").unwrap_or(label);
+        let duration_ms = *duration * 1000.0;
+        write!(labels, "{label}: \n").unwrap();
+        write!(durations, "{duration_ms:.3}\n").unwrap();
```

Comment:
Haha, now I understand why it's correctly aligned, it's just two nodes/columns, one for labels, the other for times. ^_^ If you would have created two nodes per row, then is wouldn't work (and you would have to update the values in some other way).

Following your approach, for nesting I suppose you could just add `nesting_level * "  "` spaces before each label… hacky but probably works.